server.js

```diff
@@ -1,35 +1,121 @@
+import goldenGlobesData from "./data/golden-globes.json";
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
+const GoldenGlobes = mongoose.model("GoldenGlobes", {
+  year_award: Number, 
+  nominee: String, 
+  win: Boolean
+});
+
+
+if(process.env.RESET_DB) {
+  const resetDataBase = async () => {
+    await GoldenGlobes.deleteMany();
+    goldenGlobesData.forEach(item => {
+      const newItem = new GoldenGlobes(item);
+      newItem.save();
+    })
+  }
+  resetDataBase();
+}
+
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = {
+    Welcome: "Hi! This is an open API about Golden Globes",
+    Routes: [
+      {
+        "/goldenglobes": "Shows an array of all Golden Globes objects",
+        "/goldenglobes/year/:year_award": "Displays all nominees from a specific year",
+        "/goldenglobes/nominees/:nominee": "Displays information on a specific nominee",
+        "/goldenglobes/winners/:win": "Displays an array of all the winners or the ones who did not win. True/False",
+      },
+    ],
+  }
+  res.send(endpoints)
+});
+
+app.get("/goldenglobes", async (req, res) => {
+  const allGoldenGlobes = await GoldenGlobes.find()
+  res.json(allGoldenGlobes)
+})
+
+app.get("/goldenglobes/year/:year_award", async (req, res) => {
+  try {
+    const yearAward = await GoldenGlobes.find({year_award: req.params.year_award});
+    if (yearAward) {
+      res.status(200).json({
+        success: true,
+        body: yearAward
+      });
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+          message: "No nominee found by that year"
+        }
+      });
+    }
+  } catch(error) {
+    res.status(400).json({
+      success: false,
+      body: {
+        message: "Invalid input"
+      }
+    });
+  }
+  
+});
+app.get("/goldenglobes/nominees/:nominee", async (req, res) => {
+  const { nominee } = req.params
+  try {
+    const nominees = await GoldenGlobes.find({nominee: nominee});
+    if (nominees) {
+      res.status(200).json({
+        success: true,
+        body: nominees
+      });
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+          message: "No nominees found"
+        }
+      });
+    }
+  } catch(error) {
+    res.status(400).json({
+      success: false,
+      body: {
+        message: "Invalid nominee name"
+      }
+    });
+  }
```

Comment:
You can simplify this by destructuring the body.

`app.get("/globes/", async (req, res) => {
  const { nominee, ceremony } = req.query;
  const response = {
    success: true,
    body: {}
  }
  const nomineeQuery = nominee ? nominee : /.*/;
  const ceremonyQuery = ceremony ? ceremony : {$gt: 0, $lt: 100};
  try {
    response.body = await Globe.find({nominee: nomineeQuery, ceremony: ceremonyQuery})
    res.status(200).json({
      body: response
    });
  } catch (error) {
    res.status(400).json({
      success: false,
      body: { 
        message: error
      }
    })
  }
});`

As u mentioned about If Else, mine did not work last week, but this week with this code it worked better at least😅